### ISSUE TITLE
URGENT: UNABLE to detect Navigation App in Android Auto

### ISSUE THREAD

**Issue by Android-AutoErrors** [android/car-samples#13]
UNABLE to detect navigation app in Android Auto

We are working on a Navigation application and are facing problems in getting  navigation app detected in Android Auto

Below you will see:

a) Libraries we used
b) Android Manifest file
c) automative app desc file code for Android auto
d) Different solutions we tried but didn't work

For the same we have followed the given documentation: https://developer.android.com/training/cars/navigation

a) We have used the given libraries (To check the working with both)
implementation “androidx.car.app:app:1.0.0”
implementation "androidx.car.app:app:1.0.0-rc01"

Now the issue we are currently facing is the detection of app in Android Auto (The android auto along with the desktop head unit for testing are both updated to the latest version)

b) Here is how the Android Manifest file looks:-
<service
android:name=".AndroidAuto.HelloWorldService"
android:exported="true">
            <intent-filter>

<action android:name="androidx.car.app.CarAppService"/>
                <category android:name="androidx.car.app.category.NAVIGATION"/>

                
</intent-filter>
        </service>

and the meta data is:-
<meta-data
android:name="com.google.android.gms.car.application"
android:resource="@xml/automotive_app_desc" />

c) Further the automative app desc file code for Android auto is:-
<automotiveApp xmlns:tools="http://schemas.android.com/tools">
    <uses name="template"
tools:ignore="InvalidUsesTagAttribute" />
</automotiveApp>

d) Now I would like to point out couple of things:-

1. The name="template" is not supported we have to add tools:ignore="InvalidUsesTagAttribute"
2. The multiple combination so tried includes:-

i) Simple changing template to - media/sms/notifcation (Still not detected in android auto)
ii) Further on changing the service's name to Messaging or Media and keeping template as the name in xml file the app is still not detected in android auto
iii) Now the only case the app was detected in android auto was when I switched the category to Media/Messaging in intent filter of service and keeping the automotiveApp tag name as media/sms/notifications.

So the combination in iii) point although detected by android auto doesnt fit
requirement of Navigation App. 

In conclusion the setup for Android Manifest and XML file are working fine and are getting detected by android auto in case of Media/Messaging app 
refernece taken from: https://developer.android.com/training/cars/messaging

But on the same website content of the given files present in document for Navigation: https://developer.android.com/training/cars/navigation
is not working.

Please guide us on what changes needs to be done in order to fix the same.

**Comment by Android-AutoErrors** [comment]
Hi,
I have resolved the <template> part but yet its not getting detected on the Desktop head unit. Not only Desktop Head unit it is not even getting detected on the Android Auto App when you open Customize launcher.
The issue is the same settings works fine with  
i) Change from template to media and
ii) Change from action android:name="androidx.car.app.CarAppService  to action android:name="android.media.browse.MediaBrowserService".

With the combination of above two the app is getting detected in the Android Auto App > Customise launcher as well as desktop head unit.

I also added
android.useAndroidX=true
android.enableJetifier=true

in gradle properties but no luck.

Other details you might find relevant:
Android Studio Version: 4.2.1
Android Auto is running on Device with Android 10
Google services version is: 4.3.8
Desktop head unit version is: 1.1
Build tool version is : 30.0.3
Min Sdk Version: 23
Target sdk version : 30

**Comment by mayayunx** [comment]
Hi, did you find out how to fix it? I have exactly the same issue. However yesterday for a short time, the Hello world app was detected on AA, so I tried the exact same thing on my own app, but it doesn't work. Then I reinstalled the Helloworld app again and it wasn't detected anymore, although I didn't change anything... It is really frustrating

**Comment by Android-AutoErrors** [comment]
No, still trying various combinations.  Will post the solution here once we are able to detect it in Android Auto. 

>>However yesterday for a short time, the Hello world app was detected on AA,  

Would you please share your Android Manifest file, XML file, and Gradle file which was used EXACTLY at the moment when the sample "Hello world app" was detected on AA? If it was detected only for a short time that means it has something to do with temp files/cookies,...etc. Did you try clearing these?

**Comment by Android-AutoErrors** [comment]
No, still trying various combinations.  Will post the solution here once we are able to detect it in Android Auto. 
>>However yesterday for a short time, the Hello world app was detected on AA,  
Would you please share your Android Manifest file, XML file, and Gradle file which was used EXACTLY at the moment when the sample "Hello world app" was detected on AA? If it was detected only for a short time that means it has something to do with temp files/cookies,...etc. Did you try clearing these?

    On Sunday, May 23, 2021, 05:42:06 AM EDT, mayayunx ***@***.***> wrote:  
 
 


Hi, did you find out how to fix it? I have exactly the same issue. However yesterday for a short time, the Hello world app was detected on AA, so I tried the exact same thing on my own app, but it doesn't work. Then I reinstalled the Helloworld app again and it wasn't detected anymore, although I didn't change anything... It is really frustrating

**Comment by Android-AutoErrors** [comment]
@mayayunx    what is your android auto version:  { You can access that by going to the Android Auto phone app:
From the top-left drawer menu, select settings
Scroll to the bottom and expand the "Version and Permission info" entry  }

Try with one of the below versions of android auto and let me know if it works:   i) Android Auto version reproduced on: versionName=6.0.615334-release Head unit: R-Link 2 (Renault)  ii) Android Auto version: 6.3.611324-release.   iii) Android Auto version reproduced on: DHU 2.0-mac-beta, on phone: 6.2.610914-release

**Comment by mayayunx** [comment]
> @mayayunx what is your android auto version: { You can access that by going to the Android Auto phone app:
> From the top-left drawer menu, select settings
> Scroll to the bottom and expand the "Version and Permission info" entry }
> 
> Try with one of the below versions of android auto and let me know if it works: i) Android Auto version reproduced on: versionName=6.0.615334-release Head unit: R-Link 2 (Renault) ii) Android Auto version: 6.3.611324-release. iii) Android Auto version reproduced on: DHU 2.0-mac-beta, on phone: 6.2.610914-release

Hi I tried the 3 different AA version but it all didn't work for me. Also I tried to clean the temp files and so on, but still not working. Does it work for you now? Which AA version are you using? I was using the latest version available on Google play store.

**Comment by mayayunx** [comment]
Hi, I tried with the 6.3.611324-release, downloaded Waze as a friend suggested (because then AA scans your phone for AA-enabled apps again) and connected my phone to DHU. After that, the apps (navigation, hello world and showcase) are showing up in AA for me! Super weird but it works. maybe it can help you as well

**Comment by AidanLynott** [comment]
I've been having this exact same issue for multiple days now and haven't been able to fix it. All of the sample apps once installed, despite appearing under packages (further under Version and permission info) in Android Auto, have not been appearing under Customize Launcher or on the DHU. Meanwhile, any apps I download from Google Play (waze, sygic, etc) have been working fine. It's especially frustrating because it started happening just out of the blue on Tuesday. Mayayunx's fix didn't work for me either. I don't know if it could be a phone issue (I'm using a Samsung Galaxy S9) or what, but I'd really like to be able to use these sample apps again.

**Comment by Android-AutoErrors** [comment]
@mayayunx    1) Are you saying the order in which you installed Android Auto (install first) and then install app made difference in your case?   2) Are you able to display more than one navigation apps which are not in the play store at a time in Desktop Head Unit?  3) would you please call me on my whatsapp USA number: +1-650-224-2534 (email: ***@***.***)?
We could successfully display the app in a customized launcher  (Andriod Auto Version:6.5.612134-release) and and desktop head unit for the following combinations of Android and Desktop head unit and Android auto versions.
1)   Andriod Version:9
OS Build Number: PQ3A.190605.003Pixel Experience-20190612-0118-CAF-official
Andriod Auto Version:6.5.612134-release
play services:21.18.16(100400-374723149)
Andriod DeskTop Head Unit Version:1.1
Andriod Sdk platform:31.0.2
Andriod Studio:Android Studio 4.2.1Build #AI-202.7660.26.42.7351085, built on May 10, 2021Runtime version: 11.0.8+10-b944.6842174 amd64VM: OpenJDK 64-Bit Server VM by N/AWindows 10 10.0GC: G1 Young Generation, G1 Old GenerationCores: 8Registry: external.system.auto.import.disabled=trueNon-Bundled Plugins: com.intellij.marketplace, com.thoughtworks.gauge, wu.seal.tool.jsontokotlin, org.jetbrains.kotlin

Below are the things we learned from experiments over the last 6 weeks.
i) App is not displaying in customized launcher and in the DHU with Android version 10  phone
ii) App is getting displayed with Android version 9 phone. But, if you install more than one navigation apps which are not in the play store then both apps disappears from customized launcher and desktop head unit.
iii) Whenever the above situation in ii) appears; you need to reinstall the Android OS on the phone again, then install the android auto version, and then install your navigation app in this SEQUENCE ONLY. This is weird. Don't expect this from a technology company like Google. Google experts please fix this.
    On Thursday, June 17, 2021, 04:36:48 AM EDT, mayayunx ***@***.***> wrote:  
 
 


Hi, I tried with the 6.3.611324-release, downloaded Waze as a friend suggested (because then AA scans your phone for AA-enabled apps again) and connected my phone to DHU. After that, the apps (navigation, hello world and showcase) are showing up in AA for me! Super weird but it works. maybe it can help you as well

**Comment by yjm607** [comment]
https://issuetracker.google.com/188705847
